tests/stdlib/tasynchttpserver_transferencoding.nim

```diff
@@ -1,42 +1,56 @@
+discard """
+  matrix: "--gc:arc --threads:on; --gc:arc --threads:on -d:danger; --threads:on"
+"""
+
 import httpclient, asynchttpserver, asyncdispatch, asyncfutures
 import net
 
 import std/asyncnet
 import std/nativesockets
+import std/threadpool
 
 const postBegin = """
 POST / HTTP/1.1
 Transfer-Encoding:chunked
 
 """
 
-template genTest(input, expected) =
-  var sanity = false
-  proc handler(request: Request) {.async.} =
+template genTest(input, expected: string) =
+  proc handler(request: Request, future: Future[bool]): Future[void] {.async, gcsafe.} =
```

Comment:
```suggestion
  proc handler(request: Request, future: Future[bool]) {.async, gcsafe.} =
```

Reply:
Fixed, though I don't understand why we want to remove this annotation.

Reply:
it's redundant